react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({
+      imageIndex: 0,
+      intervalID: null
+    });
+    this.handleClick = this.handleClick.bind(this);
+    this.newInterval = this.newInterval.bind(this);
+  }
+
+  handleClick(e) {
+    let newIndex;
+    if (e.target.className === 'fas fa-chevron-left') {
+      newIndex = this.state.imageIndex - 1;
+      if (newIndex < 0) {
+        newIndex = this.props.images.length - 1;
+      }
+    } else if (e.target.className === 'fas fa-chevron-right') {
+      newIndex = this.state.imageIndex + 1;
+      if (newIndex > this.props.images.length - 1) {
+        newIndex = 0;
+      }
```

Comment:
Instead of optimistically updating the index and then maybe reverting it to something else, decide what to set it to in your conditional base on its current value.

```js
if (condition is met) {
  do something
} else {
  do something else
}
```